classes/object_manipulator/manipulator.php

```diff
@@ -84,58 +84,60 @@ public function __construct($filesystem, $config) {
     /**
      * Pushes files from local file system to remote.
      *
-     * @param  array $candidatehashes content hashes to push
      */
-    public function execute($objectrecords) {
-
+    public final function execute() {
         if (!$this->manipulator_can_execute()) {
             mtrace('Objectfs manipulator exiting early');
             return;
         }
 
-        if (count($objectrecords) == 0) {
-            mtrace('No candidate objects found.');
-            return;
-        }
+        $limit = 1000; // TODO: config option?
 
-        $this->logger->start_timing();
+        while (time() <= $this->finishtime) {
+            $objects = $this->get_candidate_objects($limit);
```

Comment:
This double nest loop looks wrong to me.

If we run out of the first batch of 1000 records then shouldn't we just exit and we'll pick up the next 1000 on the next cron tick?